fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -322,74 +431,31 @@ protected String buildFromClause() throws Exception {
+     * to the appropriate parameter table alias, used by system level query.
      * @return
      */
-    protected String buildWhereClause() {
-        final String METHODNAME = "buildWhereClause";
+    protected String buildWhereClause2() {
+        final String METHODNAME = "buildWhereClause2";
         log.entering(CLASSNAME, METHODNAME);
         
-        int parameterTableAliasIndex = 1;
         StringBuilder whereClause = new StringBuilder();
-        String resolvedTableAlias;
         String whereClauseSegment;
-        boolean querySegmentProcessed = false;
                          
         whereClause.append(WHERE_CLAUSE_ROOT);
-        whereClause.append(" AND ");
         if (!this.querySegments.isEmpty()) {
             for(int i = 0; i < this.querySegments.size(); i++) {
                 SqlQueryData querySegment = this.querySegments.get(i);
                 Parameter param = this.searchQueryParameters.get(i);
-                
-                if (querySegmentProcessed) {
-                    whereClause.append(" AND ");
-                }
+
                 whereClauseSegment = querySegment.getQueryString();
-                if (!Modifier.MISSING.equals(param.getModifier())) {
-                    whereClause.append(PARAMETER_TABLE_VAR).append(parameterTableAliasIndex).append(".").append("LOGICAL_RESOURCE_ID = R.LOGICAL_RESOURCE_ID AND ");
+                if (Modifier.MISSING.equals(param.getModifier())) {
+                    whereClause.append(AND).append(whereClauseSegment);
                 }
-                resolvedTableAlias = PARAMETER_TABLE_VAR + parameterTableAliasIndex + ".";
-                whereClauseSegment = whereClauseSegment.replaceAll(PARAMETER_TABLE_ALIAS + ".", resolvedTableAlias);
-                whereClause.append(whereClauseSegment);
-                querySegmentProcessed = true;
-                parameterTableAliasIndex++;
             }
         }
-        else {
-            // When no query segments are present (such as in a search for all instances of a particular resource type),
-            // The following must be added to the WHERE clause to ensure that only the latest version of each Resource
-            // is retrieved.
-            whereClause.append("R.RESOURCE_ID = LR.CURRENT_RESOURCE_ID");
-        }
         
         log.exiting(CLASSNAME, METHODNAME);
         return whereClause.toString();
     }
-    
-    /**
-     * 
-     * @return true if this instance represents a FHIR system level search
-     */
-    protected boolean isSystemLevelSearch() {
-        return Resource.class.equals(this.resourceType);
-    }
-    
-    /**
-     * Adds the appropriate pagination clauses to the passed query string buffer, based on the type
-     * of database we're running against.
-     * @param queryString A query string buffer.
-     * @throws Exception
-     */
-    protected void addPaginationClauses(StringBuilder queryString) throws Exception {
```

Comment:
Where does this pagination logic get added now?

Reply:
i see method by this name above in `buildFromClause()`